ntoskrnl/fsrtl/fastio.c

```diff
@@ -1605,7 +1605,13 @@ FsRtlAcquireFileExclusiveCommon(IN PFILE_OBJECT FileObject,
             FilterCallbacks->PostAcquireForSectionSynchronization(&CbData, Status, CompletionContext);
```

Comment:
I've just noticed that we are missing similar callback notifications in the following other FsRtl functions:
- `FsRtlReleaseFile` (missing `FS_FILTER_RELEASE_FOR_SECTION_SYNCHRONIZATION`)
- `FsRtlAcquireFileForModWriteEx` (missing `FS_FILTER_ACQUIRE_FOR_MOD_WRITE`)
- `FsRtlReleaseFileForModWrite` (missing `FS_FILTER_RELEASE_FOR_MOD_WRITE`)
- `FsRtlAcquireFileForCcFlushEx` (missing `FS_FILTER_ACQUIRE_FOR_CC_FLUSH`)
- `FsRtlReleaseFileForCcFlush` (missing `FS_FILTER_RELEASE_FOR_CC_FLUSH`)

Reply:
I would suggest you look at these before committing this PR, because you also mention that the FS filters actually trigger the BSODs on file release (`FsRtlReleaseFile`), where there's the notification missing.

Reply:
You brought up a good point here Hermes, but I wouldn't mind that improvement having its own fresh PR.

Reply:
OK, just implemented them recently. See my new #7289 PR. :wink: